### ISSUE TITLE
Question: Possible to return to Dock on less than 19% 

### ISSUE THREAD

**Issue by Madaraxa** [rand256/valetudo#527]
Hi.
Is it possible to add an Option to let the the robot (s50 here) return to Dock at less than 19%? As my battery is getting older the robot cant finish a full run at once and its only missing like 5 more % of battery...
Thanks in advance

**Comment by stefanmd023** [comment]
No one answered you, it's a desert here. The level at which the robot returns to the dock is decided by the firmware. There could be an undocumented miio command that changes this but don't know anyone that found it. If you manage to find it, let us know!

**Comment by sca075** [comment]
In theory.. using home assistant is possible to check the battery level and send the return to dock command via MQTT. But if you have not this kind of "automations" well, valetudo isn't a firmware replacement but if you can write a script or JS app that runs at "robot" level it should be also possible to program the robot to return at dock when the battery is below the trigger. The option one is off course much more easy.

**Comment by rand256** [comment]
not quite sure what is actually asked here.

if the question is whether it's possible to change the thresholds or completely prevent the device auto-returning to dock when it decides the battery is getting low then it's a reasonable question but we unfortunately can't modify that in low level firmware.

if the question is whether it's possible to add an additional threshold to make the device return to dock at a user-specified battery level then it can be done easily but i can't really get what's the point in doing that.

**Comment by stefanmd023** [comment]
> if the question is whether it's possible to change the thresholds or completely prevent the device auto-returning to dock when it decides the battery is getting low then it's a reasonable question but we unfortunately can't modify that in low level firmware.

yes the way I understood it is this, they want their robot to finish the run because in their opinion there is enough battery left to do so without needing to go back to the dock for charging.
